src/code/z_play.c

```diff
@@ -1618,22 +1618,16 @@ s32 func_800C04D8(GlobalContext* globalCtx, s16 camId, Vec3f* arg2, Vec3f* arg3)
     return ret;
 }
 
-#ifdef NON_MATCHING
-// missing an extra stack store/load instruction pair
 s32 func_800C05E4(GlobalContext* globalCtx, s16 camId, Vec3f* arg2, Vec3f* arg3, Vec3f* arg4) {
-    u32 ret;
+    s32 ret = 0;
     s16 camIdx = (camId == -1) ? globalCtx->activeCamera : camId;
-    Camera* camera;
+    Camera* camera = globalCtx->cameraPtrs[camIdx];
     Player* player;
 
-    camera = globalCtx->cameraPtrs[camIdx];
-
-    if (1) {} // Probably necessary to match
-
-    ret = Camera_SetParam(camera, 1, arg2);
-    ret *= 2;
+    ret |= Camera_SetParam(camera, 1, arg2);
+    ret <<= 1;
```

Comment:
Could the previous function also be updated to shift instead of multiply, and maybe use the `ret = 0` + `ret |=` pattern? (assuming that's necessary here)